presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/ResourceGroupManager.java

```diff
@@ -25,4 +29,6 @@
     void addConfigurationManagerFactory(ResourceGroupConfigurationManagerFactory factory);
 
     void loadConfigurationManager() throws Exception;
+
+    List<ResourceGroup> getRootResourceGroups();
```

Comment:
I find this confusing. Currently ResourceGroup interface doesn't suggest that there has to be a root resource group. The interface itself doesn't have any indication about the structure. So it's confusing to have ResourceGroupManager interface to have an API for getRootResourceGroups (which indicate that resource groups are arranged in a tree structure). If we are adding getRootResourceGroups here, we should probably also add a getRoot in ResourceGroup interface. Especially since ResourceGroupInfo is a tree structure. Another way is to keep "root" only a concept with InternalResourceGroup, then we should name the current ResourceGroupInfo InternalResourceGroupInfo.